strawberry/arguments.py

```diff
@@ -171,8 +171,10 @@ def convert_argument(
         enum_definition: EnumDefinition = type_._enum_definition
         return convert_argument(value, enum_definition, scalar_registry, config)
 
-    if hasattr(type_, "_type_definition"):  # TODO: Replace with StrawberryInputObject
-        type_definition: TypeDefinition = type_._type_definition
+    if hasattr(
+        type_, "__strawberry_object__"
+    ):  # TODO: Replace with StrawberryInputObject
+        type_definition: StrawberryObject = type_.__strawberry_object__
```

Comment:
should we reuse the function from above?

Reply:
I want to replace it with `__strawberry_input__` in the future. What do you think?